SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc

```diff
@@ -373,14 +373,17 @@ Plot {
 		};
 	}
 
-	domainCoordinates { |size|
+	domainCoordinates { |size, mode|
 		var range, vals;
+		// step-like displays need one more x coordinate
+		var isSteplike = this.hasSteplikeDisplay(mode);
+		if (isSteplike) { size = size + 1 };
 
 		vals = if (plotter.domain.notNil) {
 			domainSpec.unmap(plotter.domain);
 		} {
 			range = domainSpec.range;
-			if (range == 0.0 or: { size == 1 }) {
+			if (range == 0.0 or: { size == 1 } and: {isSteplike.not}) {
```

Comment:
I think you need parentheses: 
```
if (range == 0.0 or: { { size == 1 } and: { isSteplike.not } })
```

otherwise it will always require stepwise.